src/get_x_eq.jl

```diff
@@ -0,0 +1,18 @@
+function get_x_eq(x_eqs::Vector{Vector{Nemo.AbstractAlgebra.RingElem}}, y_eqs::Vector{Vector{Nemo.AbstractAlgebra.RingElem}}, n::Int, m::Int, s::Int, u::Int, gens_Rjet)
+    X = Array{Nemo.fmpq_poly}(undef, 0)
+    X_eq = Array{Nemo.fmpq_poly}(undef, 0)
+    for i in 1:n
+        X = vcat(X, [Array{Nemo.fmpq_poly}(undef, 0)])
+        poly_d = SIAN.unpack_fraction(x_eqs[i][1] - x_eqs[i][2])[1]
+        for j in 0:s+1
+            if j > 0
+                poly_d = SIAN.differentiate_all(poly_d, gens_Rjet, n + m + u, j)
+            end
+            leader = gens_Rjet[i+(n+m+u)*(j+1)]
```

Comment:
I think it would be great to do something with `i + (n + m + u)  * j`, this is very hard to read and error-prone. 
I would suggest to wrap the differential ring in some sort of structure with functions to get particular states/inputs/outputs.
I made an attempt here 
https://github.com/SciML/StructuralIdentifiability.jl/blob/dev-numeric-shnumeric/src/differential_polynomials.jl
but maybe that approach is not ideal here.